ros2_socketcan/src/socket_can_receiver.cpp

```diff
@@ -171,18 +171,24 @@ CanId SocketCanReceiver::receive_fd(void * const data, const std::chrono::nanose
   // Read
   struct canfd_frame frame;
   const auto nbytes = read(m_file_descriptor, &frame, sizeof(frame));
+
   // Checks
   if (nbytes < 0) {
     throw std::runtime_error{strerror(errno)};
   }
-  if (static_cast<std::size_t>(nbytes) < sizeof(frame)) {
-    throw std::runtime_error{"read: incomplete CAN FD frame"};
+
+  if (static_cast<std::size_t>(nbytes) < sizeof(frame.can_id) + sizeof(frame.len)) {
+    throw std::runtime_error{"read: corrupted CAN frame"};
   }
-  if (static_cast<std::size_t>(nbytes) != sizeof(frame)) {
-    throw std::logic_error{"Message was wrong size"};
+
+  const auto data_length = static_cast<CanId::LengthT>(frame.len);
+  // some CAN FD frames are shorter than 64 bytes
```

Comment:
@xmfcx Let's move the check to after the point where we know we have received enough bytes to fill the `len` field.
```suggestion
  }

  if (static_cast<std::size_t>(nbytes) < sizeof(frame.can_id) + sizeof(frame.len)) {
    throw std::runtime_error{"read: corrupted CAN frame"};
  }

  const auto data_length = static_cast<CanId::LengthT>(frame.len);
  if (data_length > CANFD_MAX_DLEN) {
    throw std::runtime_error{"read: frame length is larger than max allowed CAN FD payload length"};
  }

  // some CAN FD frames are shorter than 64 bytes
```

Reply:
@xmfcx @knzo25 I don't have write access here, so someone else has to actually make the change :bow:

Reply:
applied in https://github.com/autowarefoundation/ros2_socketcan/pull/50/commits/8ca631c8d2d459fabe728e3b24ab08bcf5350b78